api/v1/nodescan/service.go

```diff
@@ -238,7 +240,31 @@ func (s *serviceImpl) getRuntimeVulns(containerRuntime *v1.GetNodeVulnerabilitie
 	return nil, nil
 }
 
-func (s *serviceImpl) GetNodeVulnerabilities(_ context.Context, req *v1.GetNodeVulnerabilitiesRequest) (*v1.GetNodeVulnerabilitiesResponse, error) {
+func (s *serviceImpl) GetNodeVulnerabilities(ctx context.Context, req *v1.GetNodeVulnerabilitiesRequest) (*v1.GetNodeVulnerabilitiesResponse, error) {
+	// If NodeInventory is empty `req.GetComponents() == nil` then fallback to v1 scanning
+	if req.GetComponents() == nil || !featureFlags.RHCOSNodeScanning.Enabled() {
```

Comment:
just curious, why not just `!featureFlags.RHCOSNodeScanning.Enabled()`?

Reply:
A request with `req.GetComponents() == nil` means that this is the legacy request (data coming not from Compliance), so I wanted this piece of code to behave exactly as before the change.

The second reason is that all the code below this line relies on `req.GetComponents()` not being nil, so this serves as panic protection.